README.md

```diff
@@ -279,15 +284,45 @@ Detects changed files that have not yet been committed. This does **not** includ
 
 Detects files that are not yet tracked by git (see `git status` manual). This does **not** include _uncommitted_ files. A new file is not _untracked_ anymore after it is added to the index.
 
+### gib.buildAll
+
+Builds all modules. Can be used to (temporarily) override the reduction of modules to build.
+
+Can be combined/useful with `gib.skipTestsForNotImpactedModules` and/or `gib.argsForNotImpactedModules`.
+
+### gib.buildDownstream
+
+By default, GIB not only builds modules that contain changes but also the modules that depend on those changed modules (donwstream modules of the changed modules).
+
+Setting this property to `false` excludes such downstream modules.
+
+GIB does _not_ evaluate Maven core `also-make-dependents` mode (`mvn -amd`).
+
+### gib.buildUpstreamMode
+
+In conjunction with `mvn -am` this property allows to control the selection of upstream modules (= modules that the modules GIB has determined to build _depend on_):
```

Comment:
So, without "-am" this does not build upstream? Why not rather ignore the "am" and always build upstream or rename to "allowBuildUpstream"?

Reply:
> Comparing how GIB handles `-am` and _not_ diretly reacts to `-amd`.

Correct.

> Why not rather ignore the "am" and always build upstream or rename to "allowBuildUpstream"?

Well, `-am` is already there and it's an option that many Maven know and that you can just active when needed.
The new property `buildUpstreamMode` on the other side is, IMO, a property you don't change all the time. It's like a more permanent configuration how you as a user want GIB to behave **if** building upstream modules (via `-am`).
So it seems more DRY to me to just re-use `-am` (as before this PR), but provide the users with a property to influence the details.
I want to avoid that users have to remember a rather clumsy GIB property when they could just use a well-known Maven property.
Am I making sense here?

Maybe it would be more clear/consistent if GIB would also react to `-amd`.
Ideas for brainstorming:

- `buildUpstream`
  - `always` (alias: `true`)
  - `derived`: only if `-am` is present
  - `never` (alias: `false`)

- `buildUpstreamScope` (or `...Scale`, or `...Extent`, or `...Range`, or ?, or stick with `buildUpstreamMode`?)
  - `changed`
  - `impacted`

- `buildDownstream`
  - `always` (alias: `true`)
  - `derived`: only if `-amd` is present
  - `never` (alias: `false`)

- `buildDownstreamScope`? I don't think we need that.

Reply:
May I kindly ask for your feedback as well, @ambition-consulting? Thank you!

Reply:
Anyone? @vackosar? I'd like to get on with this. Thanks!

Reply:
Sounds ok. When do we release 3.8?

Reply:
Thanks for your feedback!

> When do we release 3.8?

Once this PR is updated and merged, I'd say. My goal is to work on this on sunday.